client/src/main/kotlin/io/spine/examples/shareaware/client/wallet/WalletPage.kt

```diff
@@ -105,15 +115,172 @@ fun WalletPage() = Column {
                 modifier = Modifier
                     .fillMaxHeight()
             ) {
-                PrimaryButton({}, "Replenish")
+                PrimaryButton({ replenishmentState = true }, "Replenish")
             }
             Column(
                 verticalArrangement = Arrangement.Center,
                 modifier = Modifier
                     .fillMaxHeight()
             ) {
-                PrimaryButton({}, "Withdraw")
+                PrimaryButton({ withdrawalState = true }, "Withdraw")
             }
         }
+        var replenishmentIbanValue by remember { mutableStateOf("") }
+        var replenishmentAmount by remember { mutableStateOf("") }
+        if (replenishmentState) {
```

Comment:
Please work on removing this juggling with two booleans here and there. It's a tricky logic, and thus it looks very much error-prone.